### PR TITLE
Adding common configuration classes to remove duplication

### PR DESCRIPTION
**Pull request by millerts**
Depends on https://github.com/SmarterApp/TDS_Common/pull/5

### CODE REVIEW

File: service/src/main/java/tds/config/configuration/ConfigServiceApplicationConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import tds.common.configuration.RestTemplateConfiguration;
+import tds.common.web.advice.ExceptionAdvice;
+
+/**
+ * This is the base configuration class for the config microservice
+ */
+@Configuration
+@Import({ExceptionAdvice.class, RestTemplateConfiguration.class})
+public class ConfigServiceApplicationConfiguration {
```
Comment (millerts) [inline service/src/main/java/tds/config/configuration/ConfigServiceApplicationConfiguration.java]: My thinking in this class is that rather than clutter up our Application runner class that we should instead have a base configuration.  This would be where we would configure any specific beans that we don't get with auto wiring.

File: service/src/main/java/tds/config/configuration/WebConfiguration.java [hunk 2]
```diff
@@ -1,14 +0,0 @@
-package tds.config.configuration;
-
-import org.springframework.boot.web.client.RestTemplateBuilder;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.web.client.RestTemplate;
-
-@Configuration
-public class WebConfiguration {
```
Comment (millerts) [inline service/src/main/java/tds/config/configuration/WebConfiguration.java]: Moved to common

File: service/src/main/java/tds/config/web/advice/ExceptionAdvice.java [hunk 3]
```diff
@@ -1,46 +0,0 @@
-package tds.config.web.advice;
-
-import org.slf4j.Logger;
```
Comment (millerts) [inline service/src/main/java/tds/config/web/advice/ExceptionAdvice.java]: moved to common

File: service/src/test/java/tds/config/web/endpoints/AccommodationControllerIntegrationTests.java [hunk 4]
```diff
@@ -29,6 +30,9 @@
     @MockBean
     private AccommodationsService mockAccommodationsService;
 
+    @MockBean
+    private RestTemplateBuilder restTemplateBuilder;
```
Comment (millerts) [inline service/src/test/java/tds/config/web/endpoints/AccommodationControllerIntegrationTests.java]: I'm going to remove this

File: service/src/test/java/tds/config/web/endpoints/ConfigControllerIntegrationTests.java [hunk 5]
```diff
@@ -30,6 +32,7 @@
 
 @RunWith(SpringRunner.class)
 @WebMvcTest(ConfigController.class)
+@Import(ExceptionAdvice.class)
```
Comment (millerts) [inline service/src/test/java/tds/config/web/endpoints/ConfigControllerIntegrationTests.java]: The integration tests will need to import Exception advice.  This is due to how Spring WebMvcTest only loads certain things into the application context.  I feel like Spring should pull in the advice without us having to import it and it is a bug in their testing harness.

File: service/src/test/java/tds/config/web/endpoints/AccommodationControllerIntegrationTests.java (hunk 4 above)
Comment (jtreuting) [inline service/src/test/java/tds/config/web/endpoints/AccommodationControllerIntegrationTests.java]: Is this used?

File: service/src/test/java/tds/config/web/endpoints/HealthControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class HealthControllerTest {
```
Comment (jtreuting) [inline service/src/test/java/tds/config/web/endpoints/HealthControllerTest.java]: This doesn't seem related to that common changes.  Was this something you just noticed was missing while you were in there?

File: service/src/test/java/tds/config/web/endpoints/AccommodationControllerIntegrationTests.java (hunk 4 above)
Comment (millerts) [inline service/src/test/java/tds/config/web/endpoints/AccommodationControllerIntegrationTests.java]: Removing

File: service/src/test/java/tds/config/web/endpoints/HealthControllerTest.java (hunk 6 above)
Comment (millerts) [inline service/src/test/java/tds/config/web/endpoints/HealthControllerTest.java]: Yeah, I just noticed it was there and added it because I can't help myself adding test coverage.
